internal/periodic_checker/publisher.go

```diff
@@ -0,0 +1,268 @@
+	}
+
+	if err := p.kvStore.Set(ctx, taskKey, lastPublishJSON); err != nil {
+		return fmt.Errorf("failed to set last publish: %w", err)
+	}
+
+	return nil
+}
+
+func (p *PeriodicTaskPublisher) publishTask(ctx context.Context, taskType PeriodicTaskType, orgID uuid.UUID) error {
+	taskReference := PeriodicTaskReference{
+		Type:  taskType,
+		OrgID: orgID,
+	}
+
+	taskReferenceJSON, err := json.Marshal(taskReference)
+	if err != nil {
+		p.log.Errorf("Failed to marshal task reference: %v", err)
+		return err
+	}
+
+	if err := p.publisher.Publish(ctx, taskReferenceJSON); err != nil {
+		p.log.Errorf("Failed to publish task: %v", err)
+		return err
+	}
+
+	return nil
+}
+
+func (p *PeriodicTaskPublisher) Start(ctx context.Context) {
+	go func() {
+		p.syncOrganizations(ctx)
+
+		taskTicker := time.NewTicker(p.taskTickerInterval)
+		defer taskTicker.Stop()
+		orgTicker := time.NewTicker(p.orgTickerInterval)
+		defer orgTicker.Stop()
+
+		for {
+			select {
+			case <-taskTicker.C:
+				p.publishTasks(ctx)
+			case <-orgTicker.C:
+				p.syncOrganizations(ctx)
+			case <-ctx.Done():
+				p.stopAll()
+				return
+			}
+		}
+	}()
+}
+
+func (p *PeriodicTaskPublisher) syncOrganizations(ctx context.Context) {
+	p.log.Info("Syncing organizations")
+
+	orgList, status := p.orgService.ListOrganizations(ctx)
+	if status.Code < 200 || status.Code >= 300 {
+		p.log.Errorf("Failed to list organizations: %v", status)
+		return
+	}
+
+	// Track which organizations we've seen
+	organizations := make(map[uuid.UUID]bool)
+
+	for _, org := range orgList.Items {
+		orgId, err := uuid.Parse(*org.Metadata.Name)
+		if err != nil {
+			p.log.Errorf("Failed to parse organization ID %s: %v", *org.Metadata.Name, err)
+			continue
+		}
+
+		p.mu.RLock()
+		isRegistered := p.organizations[orgId]
+		p.mu.RUnlock()
+
+		if !isRegistered {
+			p.log.Infof("Registering organization %s", orgId)
+		}
+
+		organizations[orgId] = true
+	}
+
+	p.mu.Lock()
+	defer p.mu.Unlock()
+
+	for orgID := range p.organizations {
+		if !organizations[orgID] {
+			p.log.Infof("Organization %s is no longer registered, removing from tracking", orgID)
+			delete(p.organizations, orgID)
+		}
+	}
+
+	p.organizations = organizations
+}
+
+func (p *PeriodicTaskPublisher) stopAll() {
+	// Clear all organizations
+	p.mu.Lock()
+	p.organizations = make(map[uuid.UUID]bool)
+	p.mu.Unlock()
+}
+
+// Helper functions used in testing
+func (p *PeriodicTaskPublisher) getOrganizationCount() int {
+	p.mu.RLock()
+	defer p.mu.RUnlock()
+	return len(p.organizations)
+}
+
+func (p *PeriodicTaskPublisher) addOrganization(orgID uuid.UUID) {
+	p.mu.Lock()
+	defer p.mu.Unlock()
+	if p.organizations == nil {
+		p.organizations = make(map[uuid.UUID]bool)
+	}
+	p.organizations[orgID] = true
+}
+
+func (p *PeriodicTaskPublisher) hasOrganization(orgID uuid.UUID) bool {
+	p.mu.RLock()
+	defer p.mu.RUnlock()
+	return p.organizations[orgID]
+}
+
+func (p *PeriodicTaskPublisher) SetTaskTicker(interval time.Duration) {
```

Comment:
Since this is only used in the Start method and isn't going to have any effect after starting, it should really be part of the New... method only. Having it like this implies you can dynamically change it after startup.